### DISCUSSION TITLE
Add multithreading

### DISCUSSION THREAD

**Discussion by andrii29** [ansible/ansible-lint#977] [Ideas]
##### Summary

Add multithreading to app to make  lint tests much faster

##### Issue Type

- Feature Idea

##### Additional Information

I see that ansible-lint use only 1 cpu core. My tests show that cpu core is 100% loaded when use ansible-lint. With 1 core app need 2-3 min to finish linting. With multithreading linting will finish much faster!
TY!

```console
find ansible -maxdepth 1 -name "*.yml" -o -name "*.yaml"  | \
xargs grep -Rl '^- hosts: \|^  hosts: \|^- name: ' --include \*.yml --include \*.yaml | \
xargs ansible-lint -c ansible-lint.yml
```

**Comment by felixfontein** [comment]
multithreading in Python only helps when the CPU load is not 100%. Forking (multiprocessing) would help, though.
